client/me/purchases/cancel-purchase/index.jsx

```diff
@@ -59,18 +59,16 @@ class CancelPurchase extends Component {
 		purchaseListUrl: purchasesRoot,
 	};
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
+	componentDidMount() {
 		if ( ! this.isDataValid() ) {
 			this.redirect( this.props );
 			return;
 		}
 	}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.isDataValid() && ! this.isDataValid( nextProps ) ) {
-			this.redirect( nextProps );
+	componentDidUpdate( prevProps ) {
+		if ( this.isDataValid( prevProps ) && ! this.isDataValid() ) {
+			this.redirect( this.props );
```

Comment:
The `redirect()` method now always uses `this.props`. That means we can effectively remove that argument and reference `this.props` from within that method directly.

Reply:
Hmm, I tried removing the argument from `redirect()` but the page doesn't redirect when testing. 🤔 What could I be doing wrong?

Reply:
That change looks good FWIW 👍